utils/react/Hooks.js

```diff
@@ -66,4 +66,16 @@ export function useOnVisibleChange(fn) {
         observer.current?.disconnect();
         if (node) observer.current = observeVisibleChange(fn, node);
     }, []);
+}
+
+/**
+ * Hook to run a function when component scrolls.
+ *
+ * @param {function} fn
+ * @returns {function} - callback ref to be placed on target component
+ */
+export function useOnScroll(fn) {
```

Comment:
so had to go back to hook university a bit to remember how this all works. 

-  Should we be cleaning up after ourselves the way we do with onVisibleChange() and onResize().  That is, keep a ref to the listener function added, and remove it when the node is replaced, or we are unmounted.

- Should we set {passive: true} on this ?  Do we expect fn(e) {} to ever call e.preventDefault()?  I don't think so.

Reply:
would also be great to document for all 3 of these hooks what arguments get presented to fn()

Reply:
> Should we be cleaning up after ourselves the way we do with onVisibleChange() and onResize(). That is, keep a ref to the listener function added, and remove it when the node is replaced, or we are unmounted.

I don't think this is necessary - the event listener is on the element itself, and my understanding is that it will be cleaned up automatically once the element is removed from the DOM. `useOnResize()` is different as it creates a `ResizeObserver` instance that needs to call disconnect().

> Should we set {passive: true} on this ? Do we expect fn(e) {} to ever call e.preventDefault()? I don't think so.

`scroll` is passive by default, so no need to explicitly set it to true. But yes, I agree that we don't expect the callback to call preventDefault, and thus shouldn't support setting `{passive: false}`. `e.preventDefault()` actually has no effect for `scroll` events anyway, because it triggers after the scroll has already happened.

Reply:
Great!  Thanks for clarifying this!